File: web/WebFirstLab/AjaxFree/php/Timer.php
<?php
    require_once('Resultable.php');

    class Timer {
        private $resultable;

        public function __construct(Resultable $resultable) {
            $this->resultable = $resultable;
        }

        public function time() {
            $time = microtime();
            $this->resultable->result();
            
            return number_format(microtime() - $time, 7, '.', '.');
        }   
    }
?>

File: web/WebThirdLab/src/main/java/Entities/Timer.java
package Entities;

import java.text.DecimalFormat;

public class Timer {
    private final Resultable resultable;

    public Timer(Resultable resultable) {
        this.resultable = resultable;
    }

    public String time() {
        long startTime = System.nanoTime();
        resultable.result();
        long endTime = System.nanoTime();

        DecimalFormat df = new DecimalFormat("#.#");
        df.setMaximumFractionDigits(8);

        return df.format((endTime - startTime) / 1000000f).replace(",", ".");
    }
}
